### DISCUSSION TITLE
Feasibility of monochrome hi-res (512x192) mode

### DISCUSSION THREAD

**Discussion by anarsoul** [UzixLS/zx-sizif-512#51] [Q&A]
Hi,

Does CPLD have enough spare cells for implementing monochrome hi-res mode (like in Profi or Timex 2068)? Is it feasible to implement it?

**Comment by UzixLS** [comment]
Current CPLD logic elements usage is 1118 out of 1270 (88%). So my answer - may be...
But I'm not sure if this mode would be useful. There is not so much games or soft that benefit from this mode. Honestly I don't know any. May be some CP/M variant? Not sure.

**Reply by anarsoul** [reply]
I was thinking about adding hi-res support for moon rabbit (gopher browser): https://github.com/nihirash/moon-rabbit-zx

**Reply by tarhim** [reply]
FDD3000 disk drive has CP/M 2.2 and it uses Timex hires mode in terminal program, if available. It is very useful for text readability.

**Comment by amsoft78** [comment]
I've just proposed the Pull Request with Timex2048 HiRes and Hicolor modes support.
CPLD has still a lot free space. I have also experimented with two additional "CGA-like" 4 / 16 colors modes based on both Timex video pages memory, and that still fits the CPLD (almost exactly in 100%).
